frontend/src/components/Modal.js

```diff
@@ -11,78 +11,67 @@ import {
   Label,
 } from "reactstrap";
 
-export default class CustomModal extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      activeItem: this.props.activeItem,
-    };
-  }
+export const CustomModal = ({inputItem, toggle, onSave}) => {
+  const [activeItem, setActiveItem] = useState(inputItem)
 
-  handleChange = (e) => {
+  const handleChange = (e) => {
     let { name, value } = e.target;
 
     if (e.target.type === "checkbox") {
       value = e.target.checked;
     }
 
-    const activeItem = { ...this.state.activeItem, [name]: value };
-
-    this.setState({ activeItem });
+    setActiveItem({ ...activeItem, [name]: value });
```

Comment:
It's been not too long ago that in Zenput we came up with the agreement as a best practice not to use spread operators and declare each prop by it self to make it easy to understand the props that the component accepts.

Reply:
Fair enough, will change.